### PR TITLE
recursive-open-struct

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: recursive-open-struct.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    with:
+      gem: ${{vars.gem}}
+      version: ${{package.version}}
+
+  - uses: ruby/clean
+
+# test:
+#   pipeline:
+#     - runs: |
+#         ruby -e "require 'fluent/plugin/filter_concat'"
```
Comment (joshrwolf) [inline recursive-open-struct.yaml:41]: what happened here?

File: recursive-open-struct.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package:
+  name: recursive-open-struct
```
Comment (Dentrax) [inline recursive-open-struct.yaml:2]:
```suggestion
  name: recursive-open-struct
  name: ruby3.2-recursive-open-struct
```

File: recursive-open-struct.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    with:
+      gem: ${{vars.gem}}
+
+  - uses: ruby/install
+    with:
+      gem: ${{vars.gem}}
+      version: ${{package.version}}
+
+  - uses: ruby/clean
+
```
Comment (Dentrax) [inline recursive-open-struct.yaml:37]: Can we add simple import test here? Like:

```
test:
  pipeline:
    - runs: |
        ruby -e "require 'recursive-open-struct'"
```

**Review by Dentrax — CHANGES_REQUESTED** [review]

File: ruby-3.2-recursive-open-struct.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      gem: ${{vars.gem}}
+      version: ${{package.version}}
+
+  - uses: ruby/clean
+
+test:
+  pipeline:
+    - runs: |
+        ruby -e "require 'recursive-open-struct'"
+
```
Comment (Dentrax) [inline ruby-3.2-recursive-open-struct.yaml:42]: Can we add something like:
https://github.com/aetherknight/recursive-open-struct?tab=readme-ov-file#usage

```suggestion
     - name: Simple test
       runs: |
         cat <<EOF > /tmp/test.rb
         require 'recursive-open-struct'
         ros = RecursiveOpenStruct.new( { wha: { tagoo: 'siam' } } )
         ros.wha.tagoo
         EOF
         ruby /tmp/test.rb
```

**Review by Dentrax — APPROVED** [review]

**Pull request merged**